WowPacketParserModule.V8_0_1_27101/Parsers/CombatLogHandler.cs

```diff
@@ -0,0 +1,269 @@
+            packet.ReadUInt16("PlayerItemLevel", idx);
+            packet.ReadUInt16("Flags", idx);
+
+            packet.ResetBitReader();
+            packet.ReadBit("UnkBit", idx);
+        }
+
+        public static void ReadAttackRoundInfo(Packet packet, params object[] indexes)
+        {
+            var hitInfo = packet.ReadInt32E<SpellHitInfo>("HitInfo", indexes);
+
+            packet.ReadPackedGuid128("AttackerGUID", indexes);
+            packet.ReadPackedGuid128("TargetGUID", indexes);
+
+            packet.ReadInt32("Damage", indexes);
+            packet.ReadInt32("OriginalDamage", indexes);
+            packet.ReadInt32("OverDamage", indexes);
+
+            var subDmgCount = packet.ReadBool("HasSubDmg", indexes);
+            if (subDmgCount)
+            {
+                packet.ReadInt32("SchoolMask", indexes);
+                packet.ReadSingle("FloatDamage", indexes);
+                packet.ReadInt32("IntDamage", indexes);
+
+                if (hitInfo.HasAnyFlag(SpellHitInfo.HITINFO_PARTIAL_ABSORB | SpellHitInfo.HITINFO_FULL_ABSORB))
+                    packet.ReadInt32("DamageAbsorbed", indexes);
+
+                if (hitInfo.HasAnyFlag(SpellHitInfo.HITINFO_PARTIAL_RESIST | SpellHitInfo.HITINFO_FULL_RESIST))
+                    packet.ReadInt32("DamageResisted", indexes);
+            }
+
+            packet.ReadByteE<VictimStates>("VictimState", indexes);
+            packet.ReadInt32("AttackerState", indexes);
+
+            packet.ReadInt32<SpellId>("MeleeSpellID", indexes);
+
+            if (hitInfo.HasAnyFlag(SpellHitInfo.HITINFO_BLOCK))
+                packet.ReadInt32("BlockAmount", indexes);
+
+            if (hitInfo.HasAnyFlag(SpellHitInfo.HITINFO_RAGE_GAIN))
+                packet.ReadInt32("RageGained", indexes);
+
+            if (hitInfo.HasAnyFlag(SpellHitInfo.HITINFO_UNK0))
+            {
+                packet.ReadInt32("Unk Attacker State 3 1", indexes);
+                packet.ReadSingle("Unk Attacker State 3 2", indexes);
+                packet.ReadSingle("Unk Attacker State 3 3", indexes);
+                packet.ReadSingle("Unk Attacker State 3 4", indexes);
+                packet.ReadSingle("Unk Attacker State 3 5", indexes);
+                packet.ReadSingle("Unk Attacker State 3 6", indexes);
+                packet.ReadSingle("Unk Attacker State 3 7", indexes);
+                packet.ReadSingle("Unk Attacker State 3 8", indexes);
+                packet.ReadSingle("Unk Attacker State 3 9", indexes);
+                packet.ReadSingle("Unk Attacker State 3 10", indexes);
+                packet.ReadSingle("Unk Attacker State 3 11", indexes);
+                packet.ReadInt32("Unk Attacker State 3 12", indexes);
+            }
+
+            if (hitInfo.HasAnyFlag(SpellHitInfo.HITINFO_BLOCK | SpellHitInfo.HITINFO_UNK12))
+                packet.ReadSingle("Unk Float", indexes);
+
+            ReadSandboxScalingData(packet, indexes, "SandboxScalingData");
```

Comment:
This isn't correct, SMSG_ATTACKER_STATE_UPDATE has its own version for reading content tuning params which is not shared with other combat log packets

Reply:
the structure you used works for attacker_state_update and attack_swing_landed only, not anything else

Reply:
The others are using `Spellhandler.ReadSandboxScalingData`